### PR TITLE
Update the reorderable list to use the primary scroll controller when…

### LINKED ISSUE #20672: ReorderableListView should use inherited PrimaryScrollController
The ReorderableListView breaks the iOS scroll-to-top gestures by unconditionally creating a new ScrollController: https://github.com/flutter/flutter/blob/99d5ef903c54e143a95aae997b79a00782d9c640/packages/flutter/lib/src/material/reorderable_list.dart#L175

Instead, the reorderable list should prefer using the PrimaryScrollController when available in didChangeDependencies.

### PR DESCRIPTION
**Pull request by DaveShuckerow**
… it is available.

Fixes #20672

### CODE REVIEW

File: packages/flutter/lib/src/material/reorderable_list.dart [hunk 1]
```diff
@@ -230,6 +230,12 @@ class _ReorderableListContentState extends State<_ReorderableListContent> with T
     _entranceController.addStatusListener(_onEntranceStatusChanged);
   }
 
+  @override
+  void didChangeDependencies() {
+      _scrollController = PrimaryScrollController.of(context) ?? new ScrollController();
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/reorderable_list.dart]: indentation

File: packages/flutter/test/material/reorderable_list_test.dart [hunk 2]
```diff
@@ -224,6 +224,74 @@ void main() {
...
+                height: 100.0,
+                width: 100.0,
+                child: reorderableList,
+              ),
+            ),
+          );
+        }
+
+        await tester.pumpWidget(buildWithScrollController(primary));
+        SingleChildScrollView scrollView = find.byType(SingleChildScrollView).evaluate().first.widget;
```
Comment (HansMuller) [inline packages/flutter/test/material/reorderable_list_test.dart]: Is the ".first" needed here (and below)? I think there should only be one SingleChildScrollView.

**Review by HansMuller — APPROVED** [review]
LGTM modulo a few nits

File: packages/flutter/test/material/reorderable_list_test.dart (hunk 2 above)
Comment (HansMuller) [reply packages/flutter/test/material/reorderable_list_test.dart]: You can use `tester.widget<SingleChildScrollView>(find.byType(SingleChildScrollView))`

Comment (DaveShuckerow) [reply packages/flutter/test/material/reorderable_list_test.dart]: Done.

**Comment by DaveShuckerow** [comment]
I tried restarting some tasks that had been 'pending' for over a day, and they updated to being successful.

Cirrus has started duplicate tasks that are now listed as queued/pending, but since they all passed already, I am going to merge this change.

**Pull request merged**